src/hooks/useKeyboardShortcut/index.ts

```diff
@@ -0,0 +1,74 @@
+import { useCallback, useEffect, useRef } from "react";
+
+type UseKeyboardShortcutOptions = {
+  preventDefault?: boolean;
+  overrideIgnore?: boolean;
+};
+
+// Used to prevent shortcuts from being activated on input elements.
+const INPUT_ELEMENTS = ["INPUT", "TEXTAREA", "SELECT"];
+
+const useKeyboardShortcut = (
+  shortcutKeys: string[],
```

Comment:
Would love if we could get some type safety here. 
```suggestion
  shortcutKeys: (SpecialKeys | Char)[],
```
Where SpecialKeys would be `type SpecialKeys = "Control" | "Alt" | "Shift";`
And Char would be `type Char = "a" | "b" ....`
Would be a lil verbose but would give us typesafety.